Runtime/Rust/src/serialization/mod.rs

```diff
@@ -391,38 +403,50 @@ where
         self.size() + LEN_SIZE
     }
 
-    fn _serialize_chained<W: Write>(&self, dest: &mut W) -> SeResult<usize> {
-        write_len(dest, self.len())?;
-        match *self {
+    define_serialize_chained!(*SliceWrapper<'raw, T> => |zelf, dest| {
+        match zelf {
             SliceWrapper::Raw(raw) => {
-                dest.write_all(&raw)?;
-                Ok(raw.len() + LEN_SIZE)
+                serialize_byte_slice(dest, raw)
             }
             SliceWrapper::Cooked(ary) => {
-                #[cfg(target_endian = "big")]
-                todo!();
-
-                let b: &[u8] = unsafe {
```

Comment:
I can't believe I forgot to account for padding. No one probably ran into this since the types are `#[repr(packed)]` but the code was technically invalid since nothing enforced a packed representation.

Reply:
Thanks for working on this 🖤